File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/GrayscaleConversion.vb
Imports System.IO
Imports Xfinium.Pdf
Imports Xfinium.Pdf.Graphics
Imports Xfinium.Pdf.Transforms

Namespace Xfinium.Pdf.Samples
	''' <summary>
	''' GrayscaleConversion sample.
	''' </summary>
	Public Class GrayscaleConversion
		''' <summary>
		''' Main method for running the sample.
		''' </summary>
		Public Shared Function Run(input As Stream) As SampleOutputInfo()
			' Load the input file.
			Dim document As New PdfFixedDocument(input)

			Dim grayTransform As New PdfConvertToGrayTransform()
			Dim pageTransformer As New PdfPageTransformer(document.Pages(3))
			pageTransformer.ApplyTransform(grayTransform)

			Dim output As SampleOutputInfo() = New SampleOutputInfo() {New SampleOutputInfo(document, "xfinium.pdf.sample.grayscaleconversion.pdf")}
			Return output
		End Function
	End Class
End Namespace

File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/GrayscaleConversion.cs
using System;
using System.IO;
using Xfinium.Pdf;
using Xfinium.Pdf.Graphics;
using Xfinium.Pdf.Transforms;

namespace Xfinium.Pdf.Samples
{
    /// <summary>
    /// GrayscaleConversion sample.
    /// </summary>
    public class GrayscaleConversion
    {
        /// <summary>
        /// Main method for running the sample.
        /// </summary>
        public static SampleOutputInfo[] Run(Stream input)
        {
            // Load the input file.
            PdfFixedDocument document = new PdfFixedDocument(input);

            PdfConvertToGrayTransform grayTransform = new PdfConvertToGrayTransform();
            PdfPageTransformer pageTransformer = new PdfPageTransformer(document.Pages[3]);
            pageTransformer.ApplyTransform(grayTransform);

            SampleOutputInfo[] output = new SampleOutputInfo[] { new SampleOutputInfo(document, "xfinium.pdf.sample.grayscaleconversion.pdf") };
            return output;
        }
    }
}